app/controllers/listings.controller.js

```diff
@@ -44,5 +44,39 @@ exports.createListing = function(req, res, next) {
 };
 
 exports.getById = function(req, res, next) {
+    Listing.findOne({_id: req.params.listing}, function (err, listing) {
+        if (err) {
+            res.status(500).send({successful: false, text: err.message});
+        } else if (!listing) {
+            res.status(400).send({successful: false, text: 'Can not find listing :/'});
+        } else {
+            res.json(listing);
+        }
+    }).populate('user');
+};
+
+
+exports.editListing = function (req, res, next) {
+    Listing.findOne({_id: req.body.listing}, function (err, listing) {
+        if (err) {
+            res.status(500).send({successful: false, text: err.message});
+        } else if (!listing) {
+            res.status(400).send({successful: false, text: 'Can not find listing :/'});
+        } else {
+            listing.name = req.body.name;
```

Comment:
For the attributes, you should do this instead:

```
            if (req.body.name) {
                listing.name = req.body.name;
            }
            if (req.body.description) {
                listing.description = req.body.description;
            }
            if (req.body.price) {
                listing.price = req.body.price;
            }
            if (req.body.imageUrl) {
                listing.imageUrl = req.body.imageUrl;
            }
```
That way, any attributes that are left out won't be overwritten by `null`.